src/test/java/io/stargate/sgv2/jsonapi/api/v1/util/DataApiResponseValidator.java

```diff
@@ -172,6 +172,17 @@ public <T extends APIException> DataApiResponseValidator hasSingleApiError(
     return validator;
   }
 
+  public <T extends APIException> DataApiResponseValidator mayHasSingleApiError(
```

Comment:
nit: change mayHasSingleApiError -> mayHaveSingleApiError

Also, if errorCode != null call hasSingleApiError  to remove code duplication

Reply:
fixed